wordle-web/src/scripts/wordsService.ts

```diff
@@ -7,9 +7,105 @@ export abstract class WordsService {
     return this.#words.includes(word)
   }
 
-  static validWords(): Array<string> {
-    //Todo
-    return new Array<string>()
+  static getAll() {
+    let all = Array()
+    for (const word of this.#words) {
+      all.push(word)
+    }
+    return all
+  }
+
+  static currentWords(curGuess: string[], currentList: string[]) {
+    //key:
+    //  letter[0] = char
+    //  letter[1] = status
+    //  letter[2] = index
+    let wordList = Array()
+    //let wordList = this.#words
+    for (const word of currentList) {
+      let add = true
+      let split = word.split('')
+      for (let i = 0; i < curGuess.length; i++) {
+        if (split[i] != curGuess[i]) {
+          add = false
+        }
+      }
+      if (add == true) {
+        wordList.push(word)
+      }
+    }
+    /*console.log("PressCheck")
+    console.log(currentList)
+    console.log(curGuess.length)
+    console.log(curGuess)*/
+    console.log(wordList)
+    return wordList
+  }
+
+  static validWords(knowns: string[][]): Array<string> {
+    //key:
```

Comment:
You can simplify this function a lot by doing something along these lines

static validWords(word: string): Array<string> {
    return this.#words
      .filter((w) => w.startsWith(word))
      .sort()
      .slice(0, 500)
  }

Reply:
Thank you for the code, I will give that a try later.  Right now I want to make sure everything else is looking good before making any alterations to the abomination that I have given form.